app/src/main/java/io/github/sumukhshiv/mdbsocials/DetailScreenActivity.java

```diff
@@ -0,0 +1,253 @@
+        textViewEventNameDetail.setText(social.nameOfEvent);
+        emailDetail.setText(social.emailOfHost);
+        buttonInterestedDetail.setText(Integer.toString(social.numberIntersted));
+        textViewDescription.setText(social.description);
+        class DownloadFilesTask extends AsyncTask<String, Void, Bitmap> {
+            protected Bitmap doInBackground(String... strings) {
+                try {return Glide.
+                        with(getApplicationContext()).
+                        load(strings[0]).
+                        asBitmap().
+                        into(100, 100). // Width and height
+                        get();}
+                catch (Exception e) {return null;}
+            }
+
+            protected void onProgressUpdate(Void... progress) {}
+
+            protected void onPostExecute(Bitmap result) {
+                imageViewDetail.setImageBitmap(result);
+            }
+        }
+
+        //FirebaseOptions opts = FirebaseApp.getInstance().getOptions();
+        //Log.i(TAG, "Bucket = " + opts.getStorageBucket());
+
+        FirebaseStorage.getInstance().getReferenceFromUrl("gs://mdbsocials-e2598.appspot.com").child(social.image).getDownloadUrl().addOnSuccessListener(new OnSuccessListener<Uri>() {
+            @Override
+            public void onSuccess(Uri uri) {
+                (new DownloadFilesTask()).execute(uri.toString());
+//                Log.d("ye", uri.toString());
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception exception) {
+//                Log.d("sad", exception.toString());
+            }
+        });
+//        Glide.with(this)
+//                .load("http://assets.pokemon.com/assets/cms2/img/pokedex/full/" + pokemon.number + ".png")
+//                .into(image);
+    }
+
+    private void addInterested() {
+
+        //NOTE, also add self email to people interested arraylist
+        final String socialKey = social.image.substring(0, social.image.length() - 4);
+        final DatabaseReference ref = FirebaseDatabase.getInstance().getReference("/socials").child(socialKey);
+
+        ref.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                int numberIntersted = dataSnapshot.child("numberIntersted").getValue(Integer.class);
+                numberIntersted++;
+                buttonInterestedDetail.setText(Integer.toString(numberIntersted));
+                addInterestedToDatabase(numberIntersted, socialKey);
+                ArrayList<String> a = social.peopleInterested;
+                if (a != null && !a.contains(firebaseUser.getUid() + " hello")) {
+                    a.add(firebaseUser.getUid());
+                    social.peopleInterested = a;
+                    ref.child("peopleInterested").setValue(a);
+                }
```

Comment:
I think you can use .push to do this, and to get the number of elements you can use dataSnapshot.getChildrenCount